crates/aptos/src/governance/delegation_pool.rs

```diff
@@ -0,0 +1,296 @@
+/// Also check if partial governance voting is enabled for delegation pool. If not, send a
+/// transaction to enable it.
+async fn delegation_pool_governance_precheck(
+    txn_options: &TransactionOptions,
+    pool_address: AccountAddress,
+) -> CliTypedResult<Option<TransactionSummary>> {
+    let client = &txn_options
+        .rest_options
+        .client(&txn_options.profile_options)?;
+    if !is_partial_governance_voting_enabled(client).await? {
+        return Err(CliError::CommandArgumentError(
+            "Partial governance voting feature flag is not enabled".to_string(),
+        ));
+    };
+    if !is_delegation_pool_partial_governance_voting_enabled(client).await? {
+        return Err(CliError::CommandArgumentError(
+            "Delegation pool partial governance voting feature flag is not enabled".to_string(),
+        ));
+    };
+    if is_partial_governance_voting_enabled_for_delegation_pool(client, pool_address).await? {
+        Ok(None)
+    } else {
+        println!("Partial governance voting for delegation pool {} hasn't been enabled yet. Enabling it now...",
+                 pool_address);
+        let txn_summary = txn_options
+            .submit_transaction(
+                aptos_stdlib::delegation_pool_enable_partial_governance_voting(pool_address),
+            )
+            .await
+            .map(TransactionSummary::from)?;
+        Ok(Some(txn_summary))
+    }
+}
+
+async fn is_partial_governance_voting_enabled_for_delegation_pool(
+    client: &Client,
+    pool_address: AccountAddress,
+) -> CliTypedResult<bool> {
+    let response = client
+        .view(
+            &ViewRequest {
+                function: "0x1::delegation_pool::partial_governance_voting_enabled"
+                    .parse()
+                    .unwrap(),
+                type_arguments: vec![],
+                arguments: vec![serde_json::Value::String(pool_address.to_string())],
+            },
+            None,
+        )
+        .await?;
+    Ok(response.inner()[0].as_bool().unwrap())
+}
+
+async fn get_remaining_voting_power(
+    client: &Client,
+    pool_address: AccountAddress,
+    voter_address: AccountAddress,
+    proposal_id: u64,
+) -> CliTypedResult<u64> {
+    let response = client
+        .view(
+            &ViewRequest {
+                function: "0x1::delegation_pool::calculate_and_update_remaining_voting_power"
+                    .parse()
+                    .unwrap(),
+                type_arguments: vec![],
+                arguments: vec![
+                    serde_json::Value::String(pool_address.to_string()),
+                    serde_json::Value::String(voter_address.to_string()),
+                    serde_json::Value::String(proposal_id.to_string()),
+                ],
+            },
+            None,
+        )
+        .await?;
+    Ok(response.inner()[0].as_str().unwrap().parse().unwrap())
```

Comment:
Is there a way we can wrap around this using `?` instead of `unwrap()` to avoid any sort of panic?